tracer/test/Datadog.Trace.DuckTyping.Tests/Fields/ReferenceType/ReferenceTypeFieldTests.cs

```diff
@@ -50,6 +50,9 @@ public void StaticReadonlyFields(string obscureObjectName)
             var duckAbstract = obscureObject.DuckCast<ObscureDuckTypeAbstractClass>();
             var duckVirtual = obscureObject.DuckCast<ObscureDuckTypeVirtualClass>();
 
+            ref var originalObject = ref ((IDuckType)duckInterface).GetInternalDuckTypedInstance<object>();
+            Assert.NotNull(originalObject);
```

Comment:
You can assert more than that, right? Can't we assert that it _is_ `obscureObject`?

Reply:
true! done in [7477206](https://github.com/DataDog/dd-trace-dotnet/pull/5216/commits/747720610962dbe522b074ffb123f7b407ebb08a)